jx_sqlite/setop_table.py

```diff
@@ -464,7 +468,11 @@ def _make_sql_for_one_nest_in_set_op(
                 for select_index, s in enumerate(selects):
                     sql_select = index_to_sql_select.get(select_index)
                     if not sql_select:
-                        select_clause.append(selects[select_index])
+                        sql = selects[select_index]
+                        if nested_path != "." and parent_alias not in sql and alias not in sql:
```

Comment:
This fancy condition points to a deeper problem:  We do not want SQL for the sibling records.  It would be nice to know the nested path for each item in `selects` so we have simpler logic (and correct logic) to switch on.  For example, suppose `selects` was a list of `(nested_path, sql)` tuples rather than just sql:

    nestpath, sql = selects[select_index]
    if parent_alias!=nestpath:
        select_clause.append("NULL AS "+_make_column_name(select_index)) 
    ....

So we can look at where `selects` is constructed as `sql_selects` elsewhere:

                                    sql_selects.append(unsorted_sql + " AS " + column_alias)
                                    index_to_column[column_number] = nested_doc_details['index_to_column'][column_number] = ColumnMapping(
                                        push_name=s.name,
                                        push_column_name=s.name.replace("\\.", "."),
                                        push_column=si,
                                        push_child=column.name,
                                        pull=get_column(column_number),
                                        sql=unsorted_sql,
                                        type=json_type,
                                        nested_path=[nested_path]
                                        # fake the real nested path, we only look at [0] anyway
                                    )

What'ss interesting is `sql_selects` and `index_to_column` are always constructed together; and since `index_to_column` has the `nested_path` we need; maybe we can just add the sql_select as yet-another-property to the `ColumnMapping`;  
actually, the `unsorted_sql` is already there, so maybe we just add the `column_alias` to the `ColumnMapping`.